.github/workflows/node.js.yml

```diff
@@ -0,0 +1,34 @@
+name: Node.js CI
+on:
+  push:
+    branches:
+      - master
+  pull_request:
+    branches:
+      - master
+
+env:
+  FORCE_COLOR: 1
+
+jobs:
+  build:
+    runs-on: ubuntu-latest
+    steps:
+    - uses: actions/checkout@v2
+    - uses: actions/setup-node@v1
+      with:
+        node-version: 14
+    - name: Get npm cache directory
+      id: npm-cache-dir
+      run: echo "::set-output name=dir::$(npm root)"
+    - uses: actions/cache@v2
+      id: npm-cache
+      with:
+        path: ${{ steps.npm-cache-dir.outputs.dir }}
+        key: ${{ runner.os }}-node-${{ hashFiles('**/package.json') }}
+        restore-keys: ${{ runner.os }}-node-
+    - if: steps.npm-cache.outputs.cache-hit != 'true'
+      run: npm ci
+    - run: npm install -g grunt-cli
```

Comment:
Does `npm test` work on GitHub actions? As if it does that would save having to install Grunt CLI globally each time.

I vaguely remember travis had some issue with it a long ago which required installing Grunt globally so just wondering if GitHub actions has the same issue or if this can be removed.

Reply:
https://docs.github.com/en/actions/guides/building-and-testing-nodejs

From the documentation, they are not installing grunt and running npm test.  So its possible.

Reply:
Yeah, it does, but in this case is just an alias for `grunt test` https://github.com/samclarke/SCEditor/blob/b63068c41872bb121adac093b82367fc152aec5f/package.json#L42

Reply:
npm adds` node_modules/.bin` to the PATH when executing scripts so as long as GitHub actions isn't changing anything `npm test` should work without having to install grunt globally.

Reply:
Just thought, it might also be worth adding `timeout-minutes` set to something like 5 minutes just in case any code ever contains an infinite loop or something like that.

Reply:
wut, the default is six hours? That seems excessive. I'll add a limit of ten minutes to mach Travis.

Reply:
I know, shocked me too when I saw that.